lib/streamlit/type_util.py

```diff
@@ -1074,3 +1075,55 @@ def infer_vegalite_type(data: Series[Any]) -> Union[str, Tuple[str, List[Any]]]:
         #     stacklevel=1,
         # )
         return "nominal"
+
+
+E1 = TypeVar("E1", bound=Enum)
+E2 = TypeVar("E2", bound=Enum)
+
+ALLOWED_ENUM_COERCION_CONFIG_SETTINGS = ("off", "nameOnly", "nameAndValue")
+
+
+def coerce_enum(from_enum_value: E1, to_enum_class: Type[E2]) -> E1 | E2:
+    """Attempt to coerce an Enum value to another EnumMeta.
+
+    An Enum value of EnumMeta E1 is considered coercable to EnumType E2
+    if the EnumMeta __qualname__ match and the names of their members
+    match as well. (This is configurable in streamlist configs)
+    """
+    if not isinstance(from_enum_value, Enum):
+        raise ValueError("Expected an Enum")
+    if not isinstance(to_enum_class, EnumMeta):
+        raise ValueError("Expected an EnumMeta (i.e. the class of an Enum)")
```

Comment:
nit: would be good to mention whether we expect an `Enum`/`EnumMeta` as the first or second argument in the error messages since the two messages are so similar